ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_linalg.py

```diff
@@ -606,3 +606,52 @@ def test_paddle_matrix_power(
         x=x[0],
         n=n,
     )
+
+
+# cond
+@st.composite
+def _get_dtype_and_matrix_non_singular(draw, dtypes):
+    while True:
+        matrix = draw(
+            helpers.dtype_and_values(
+                available_dtypes=dtypes,
+                min_value=-10,
+                max_value=10,
+                min_num_dims=2,
+                max_num_dims=2,
+                min_dim_size=1,
+                max_dim_size=5,
+                shape=st.tuples(st.integers(1, 5), st.integers(1, 5)).filter(
+                    lambda x: x[0] == x[1]
+                ),
+                allow_inf=False,
+                allow_nan=False,
+            )
+        )
+        if np.linalg.det(matrix[1][0]) != 0 and matrix_is_stable(matrix[1][0]):
```

Comment:
The matrix_is_stable can be used in body of the function call using `matrix_is_stable(x[0])`

Reply:
> matrix_is_stable(x[0])

how can this be done, not sure how to move about it since all tests are failing after i made some changes

Reply:
You need to use assume to filter only when matrix is stable `assume(matrix_is_stable(x[0]))`